src/systolic_array/tensor.h

```diff
@@ -217,8 +218,11 @@ class TensorIndexIterator {
       int stridesOutBound = (offset - bound) / stride;
       // The number of strides this dimension can take.
       int stridesThisDim = bound / stride + 1;
-      int carry = (stridesOutBound / stridesThisDim + 1) * nextStride;
+      int carry =
+              std::ceil(stridesOutBound * 1.0 / stridesThisDim) * nextStride;
       int newOffset = (stridesOutBound % stridesThisDim - 1) * stride;
+      if (newOffset < 0)
```

Comment:
std::max(0, bound)

Reply:
This is actually not sufficient if `newOffset` is greater than 0.